crates/router/tests/connectors/iatapay.rs

```diff
@@ -0,0 +1,144 @@
+            expires: 60 * 5,
+        }),
+        _ => None,
+    }
+}
+
+static CONNECTOR: IatapayTest = IatapayTest {};
+
+fn get_default_payment_info() -> Option<utils::PaymentInfo> {
+    Some(utils::PaymentInfo {
+        address: Some(PaymentAddress {
+            billing: Some(api::Address {
+                address: Some(api::AddressDetails {
+                    first_name: Some(Secret::new("first".to_string())),
+                    last_name: Some(Secret::new("last".to_string())),
+                    line1: Some(Secret::new("line1".to_string())),
+                    line2: Some(Secret::new("line2".to_string())),
+                    city: Some("city".to_string()),
+                    zip: Some(Secret::new("zip".to_string())),
+                    country: Some(api_models::enums::CountryAlpha2::NL),
+                    ..Default::default()
+                }),
+                phone: Some(api::PhoneDetails {
+                    number: Some(Secret::new("1234567890".to_string())),
+                    country_code: Some("+91".to_string()),
+                }),
+            }),
+            ..Default::default()
+        }),
+        access_token: get_access_token(),
+        ..Default::default()
+    })
+}
+
+fn payment_method_details() -> Option<types::PaymentsAuthorizeData> {
+    Some(types::PaymentsAuthorizeData {
+        router_return_url: Some("https://hyperswitch.io".to_string()),
+        webhook_url: Some("https://hyperswitch.io".to_string()),
+        currency: enums::Currency::EUR,
+        ..PaymentAuthorizeType::default().0
+    })
+}
+
+// Cards Positive Tests
+
+// Creates a payment checking if its status is "requires_customer_action" for redirectinal flow
+#[actix_web::test]
+async fn should_only_create_payment() {
+    let response = CONNECTOR
+        .authorize_payment(payment_method_details(), get_default_payment_info())
+        .await
+        .expect("Authorize payment response");
+    assert_eq!(response.status, enums::AttemptStatus::AuthenticationPending);
+}
+
+//refund on an unsuccessed payments
+#[actix_web::test]
+async fn should_fail_for_refund_on_unsuccessed_payment() {
+    let response = CONNECTOR
+        .authorize_payment(payment_method_details(), get_default_payment_info())
+        .await
+        .expect("Authorize payment response");
+    let refund_response = CONNECTOR
+        .refund_payment(
+            get_connector_transaction_id(response.response).unwrap(),
+            Some(types::RefundsData {
+                refund_amount: response.request.amount,
+                ..utils::PaymentRefundType::default().0
+            }),
+            get_default_payment_info(),
+        )
+        .await
+        .unwrap();
+    assert_eq!(
+        refund_response.response.unwrap_err().code,
+        "BAD_REQUEST".to_string(),
+    );
+}
+
+// Refunds a payment with refund amount higher than payment amount.
+#[actix_web::test]
+async fn should_fail_for_refund_amount_higher_than_payment_amount() {
```

Comment:
can you add tests for refund_sync and payment_sync also if it is possible

Reply:
@anji-reddy-j can u please pick this up